src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -264,7 +267,10 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, TagHelperProper
                     Suffix = "\"",
                 };
 
-                changeNode.Children.Add(new CSharpExpressionAttributeValueIntermediateNode());
+                changeNode.Children.Add(new CSharpExpressionAttributeValueIntermediateNode()
+                {
+                    Source = node.Source
```

Comment:
Do we know what the effect is of making two pieces of code have the same line mapping? What happens if you try to set a breakpoint?

Reply:
You can simulate these things btw by just making a regular C# file with pragmas if you want to test how the debugger behaves.

Reply:
I tried simulating a bunch of things in a regular cs file. If I put a breakpoint on the line that maps to two locations in the generated code, it only breaks on the first location unless the contents of the two locations are the same. In our case, the contents of valueNode and changeNode are different. So, I guess we need to split the expression in to multiple lines if we want correct debugging.

Reply:
I think it would be OK to just make one of the nodes debuggable.